File: p1.js
// ## Problem1 
// Custom Sort String (https://leetcode.com/problems/custom-sort-string/)

// Time Complexity : O(N x count)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes

const customSortString = function(S, T) {
    let map = new Map(), res = '';
    for(let c of T) {
        map.set(c, (map.get(c) || 0) + 1);
    }
    for(let c of S) {
        if(map.has(c)) {
            while(map.get(c) > 0) {
                res += c; 
                map.set(c, map.get(c) - 1 )
            }
            map.delete(c);
        }
    }
    for(let [k,v] of map) {
        while(map.get(k) > 0) {
            res += k;
            map.set(k, map.get(k) - 1)
        }
    }
    return res;
};

File: P1.py
## Problem1 
# Custom Sort String (https://leetcode.com/problems/custom-sort-string/)
# TC : O(max(len(order), len(s)))
# SC : O(1)
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        d = defaultdict(int)
        res = ''
        for i in s:
            d[i] += 1
        for j in order:
            if j in d :
                while d[j]>0:
                    res += j
                    s = s.replace(j, "", 1)
                    d[j] -= 1
        return res+s
        # print(d, res, s)
                
        